src/workerd/api/sockets.c++

```diff
@@ -329,6 +329,24 @@ void Socket::handleProxyStatus(
   result.markAsHandled(js);
 }
 
+void Socket::handleProxyStatus(jsg::Lock& js, kj::Promise<kj::Maybe<kj::Exception>> connectResult) {
+  // It's kind of weird to take a promise that resolves to a Maybe<Exception> but we can't just use
+  // a Promise<void> and put our logic in the error handler because awaitIo's errorFunc isn't passed
+  // the jsg::Lock, which we need to have in the error case.
```

Comment:
Note for later: should we update `awaitIo` so that it *does* pass the `jsg::Lock`?

Reply:
we should also update it to be awaitable

Reply:
Yeah this sounds like a bug in `awaitIo` that we should just fix instead of work around.

Reply:
It's a little bit more than I had mentioned previously -- currently the only code path that passes a `jsg::Lock` is if your awaited promise resolves to a non-void result *and* if you're on the success callback. Even if I added a `jsg::Lock` to the error callback that wouldn't help in this particular case since I had been trying to use a `Promise<void>`, and as of today both the success and error callbacks aren't given a `jsg::Lock` in that case.

I'll leave it as a cleanup for now rather than boiling the ocean on all the `awaitIo` overloads.